back/config/schemas/settings.schema.json.erb

```diff
@@ -502,10 +502,19 @@
             "description": "The boundary of the municipality/organisation is specified by its Relation ID in OpenStreetMap (OSM). You can find the Relation ID at https://www.openstreetmap.org. Search for the place, click on the correct result and you'll find the Relation ID if the result is a relation (e.g. \"Relation: Knokke-Heist (4569)\"). If there is no relation defined for the place (e.g. Scheveningen), search for the next largest administrative territory (e.g. The Hague)."
           },
           "esri_integration": {
-            "type": "boolean",
-            "default": false,
             "title": "Esri Integration",
-            "description": "Enable Esri integration for maps. In development. DO NOT ENABLE ON COMMERCIAL PLATFORM!"
+            "description": "Esri integration for maps. In development. DO NOT ENABLE ON COMMERCIAL PLATFORM!",
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+              "enabled": { "type": "boolean", "default": false},
+              "api_key": {
+                "title": "API Key",
+                "description": "The API key for the Esri integration. (TODO: add more useful information)",
+                "type": "string",
+                "private": true
```

Comment:
@alexander-cit I've added `private: true`, rebuilt BE and FE, restarted and any value entered for the api_key is still shown in the response body of GET app_configuration. What am I missing here?

Reply:
But then how does [this](https://github.com/CitizenLabDotCo/citizenlab/blob/eda81d0bfdff3e20b0ec9fac2a4d9546e8017334/back/config/schemas/settings.schema.json.erb#L727-L733) work?
```
"properties": {
    "allowed": { "type": "boolean", "default": true},
    "enabled": { "type": "boolean", "default": true},
    "user_token": {
      "title": "User Token",
      "description": "The user token is generated from the tenant's typeform profile. Survey results are only downloadable for this token.",
      "type": "string",
      "private": true,
      "default": "<%= ENV.fetch('DEFAULT_TYPEFORM_USER_TOKEN', '') %>"
    }
  }
```

Reply:
But don't worry too much, as apparently the API is read-only and the implications of exposing the key have been discussed.